src/point-cloud-layer.ts

```diff
@@ -0,0 +1,219 @@
+import {
+    CompositeLayer,
+    CompositeLayerProps,
+    DefaultProps,
+    GetPickingInfoParams,
+    Layer,
+    LayersList,
+    assert,
+    Unit,
+    Material,
+  } from "@deck.gl/core/typed";
+import { PointCloudLayer } from "@deck.gl/layers/typed";
+import type { PointCloudLayerProps } from "@deck.gl/layers/typed";
+import * as arrow from "apache-arrow"
+import * as ga from "@geoarrow/geoarrow-js"
+import {
+    assignAccessor,
+    extractAccessorsFromProps,
+    getGeometryVector,
+    invertOffsets,
+} from "./utils.js"
+// TODO which accessors are actually needed for a pointcloud layer
+import {
+    GeoArrowExtraPickingProps,
+    computeChunkOffsets,
+    getPickingInfo,
+} from "./picking.js"
+import { ColorAccessor, FloatAccessor, GeoArrowPickingInfo } from "./types.js";
+import { EXTENSION_NAME } from "./constants.js";
+import { validateAccessors } from "./validate.js";
+import { defaultPoolSize } from "threads/dist/master/implementation.browser.js";
+import { Point } from "@geoarrow/geoarrow-js/dist/type.js";
+import { getPointChild } from "@geoarrow/geoarrow-js/dist/child.js";
+
+/* All properties supported by GeoArrowPointCloudLayer */
+export type GeoArrowPointCloudLayerProps = Omit<
+    PointCloudLayerProps<arrow.Table>, // TODO is this still an Arrow Table or is it another datatype? An arrow table is a vector/array like value so is it the same?
+    | "omitted properties here"
+    > &
+    _GeoArrowPointCloudLayerProps &
+    CompositeLayerProps;
+
+/* All properties added by GeoArrowPointCloudLayer */
+type _GeoArrowPointCloudLayerProps = {
+    // data
+    data: arrow.Table,
+
+    /**
+     * If `true`, validate the arrays provided (e.g. chunk lengths)
+     * @default true
+     */
+    _validate?: boolean;
+
+    /**
+     * 
+     * The units of the point size, one of `'meters'`, `'common'`, and `'pixels'`.
+     * @default 'pixels'
+     */
+    sizeUnits?: Unit;
+
+    /**
+     * Global radius of all points, in units specified by `sizeUnits`
+     * @default 10
+     */
+    pointSize: number;
+
+    /**
+     * Material settings for lighting effect.
+     *
+     * @default true
+     * @see https://deck.gl/docs/developer-guide/using-lighting
+     */
+    material?: Material;
+
+    /** 
+    * Center position accessor.
+    * If not provided, will be inferred by finding a column with extension type
+    * `"geoarrow.point"` or `"geoarrow.multipoint"`.
+    */
+    getPosition?: ga.vector.PointVector | ga.vector.MultiPointVector;
```

Comment:
In this layer we only accept points, not multipoints.
```suggestion
    /** 
    * Center position accessor.
    * If not provided, will be inferred by finding a column with extension type
    * `"geoarrow.point"`.
    */
    getPosition?: ga.vector.PointVector;
```